core/src/test/java/io/grpc/internal/ManagedChannelImplTest.java

```diff
@@ -4476,6 +4476,87 @@ public String getDefaultScheme() {
     }
   }
 
+  @Test
+  public void testApplyDefaultServiceConfigIfInitialNameResolutionFails() throws Exception {
+    LoadBalancerRegistry.getDefaultRegistry().register(mockLoadBalancerProvider);
+    timer.forwardNanos(1234);
+    channelBuilder.maxTraceEvents(10);
+    try {
+      Status resolutionError = Status.UNAVAILABLE
+              .withDescription("Initial Name Resolution error, using default service config");
+      FakeNameResolverFactory nameResolverFactory = new FakeNameResolverFactory
+              .Builder(expectedUri).setServers(Collections.singletonList(new
+              EquivalentAddressGroup(socketAddress))).setError(resolutionError).build();
+      channelBuilder.nameResolverFactory(nameResolverFactory);
+      Map<String, Object> defaultServiceConfig =
+              parseConfig("{\"methodConfig\":[{"
+                      + "\"name\":[{\"service\":\"SimpleService1\"}],"
+                      + "\"waitForReady\":true}]}");
+
+      channelBuilder.defaultServiceConfig(defaultServiceConfig);
+      createChannel(true);
+      int prevSize = getStats(channel).channelTrace.events.size();
+
+      //assertThat(getStats(channel).channelTrace.events).hasSize(prevSize);
+      assertThat(getStats(channel).channelTrace.events.get(prevSize - 1))
+              .isEqualTo(new ChannelTrace.Event.Builder()
+              .setDescription("Initial Name Resolution error, using default service config")
+              .setSeverity(ChannelTrace.Event.Severity.CT_ERROR)
+              .setTimestampNanos(timer.getTicker().read())
+              .build());
+
+      FakeNameResolverFactory.FakeNameResolver resolver = nameResolverFactory.resolvers.get(0);
+
+      prevSize = getStats(channel).channelTrace.events.size();
+      resolver.listener.onError(resolutionError);
+
+      assertThat(getStats(channel).channelTrace.events).hasSize(prevSize);
```

Comment:
You are asserting that no new events got logged from onError. Can you explain what that should be the case?

Reply:
As per your previous comments - Our event here of interest would be the last event in the array, and prevSize - 1 is not it.

Reply:
moved trace event call after onError and assert it.

Reply:
The channel trace event for "Failed to resolve name" on a 2nd onError did  not log again because of the lastResolutionState already being an ERROR and the below check in _handleErrorInSyncContext_. 

```
if (lastResolutionState != ResolutionState.ERROR) {
        channelLogger.log(ChannelLogLevel.WARNING, "Failed to resolve name: {0}", error);
```

So the events count not increasing is the correct behavior for a consecutive onError call then.